components/script/dom/bindings/str.rs

```diff
@@ -208,6 +208,114 @@ impl DOMString {
         self.0.truncate(last_non_whitespace);
         let _ = self.0.splice(0..first_non_whitespace, "");
     }
+
+    /// A valid date string should be YYYY-MM-DD
+    /// YYYY must be four or more digits, MM and DD both must be two digits
+    /// https://html.spec.whatwg.org/multipage/#valid-date-string
+    pub fn is_valid_date_string(&mut self) -> bool {
+        match self.parse_date_component(&*self.0) {
+            Ok(_)  => true,
+            Err(_) => false
+        }
+    }
+
+    /// A valid month string should be "YYYY-MM"
+    /// YYYY must be four or more digits, MM both must be two digits
+    /// https://html.spec.whatwg.org/multipage/#valid-month-string
+    pub fn is_valid_month_string(&mut self) -> bool {
+        match self.parse_month_component(&*self.0) {
+            Ok(_)  => true,
+            Err(_) => false
+        }
+    }
+
+    /// https://html.spec.whatwg.org/multipage/#parse-a-month-component
+    fn parse_month_component(&self, value: &str) -> Result<(u32, u32), ()> {
```

Comment:
Still not an Rust expert yet.
However, I'm wondering if it's more appropriate to use `Option` here?
How do @KiChjang and @tigercosmos think?

Reply:
I need to return `Err()`, so I use `Result`

Reply:
@CYBAI It can, and in fact, syntactically there's really not a difference between `Option` and `Result`. However, _semantically_ they are different - one indicates that the function may return something nullable, and the other represents whether an operation passed or failed.

Reply:
I will note that this function is really awkward, since it doesn't use the self parameter at all.

Reply:
@KiChjang Yeah, thanks for the clarification. Indeed, we're not checking _nullable_ here; it's more correct to use `Result`.